cli/src/migrate.js

```diff
@@ -0,0 +1,354 @@
+'use strict';
+const chalk = require('chalk');
+const r = require('rethinkdb');
+const Promise = require('bluebird');
+const argparse = require('argparse');
+const child_process = require('child_process');
+const runSaveCommand = require('./schema').runSaveCommand;
+const config = require('./utils/config');
+const interrupt = require('./utils/interrupt');
+const change_to_project_dir = require('./utils/change_to_project_dir');
+const start_rdb_server = require('./utils/start_rdb_server');
+
+const VERSION_2_0 = [ 2, 0, 0 ];
+
+function run(cmdArgs) {
+  const options = processConfig(cmdArgs);
+  interrupt.on_interrupt(() => teardown());
+  return Promise.resolve().bind({ options })
+    .then(setup)
+    .then(validateMigration)
+    .then(makeBackup)
+    .then(renameUserTables)
+    .then(moveInternalTables)
+    .then(renameIndices)
+    .then(rewriteHzCollectionDocs)
+    .then(exportNewSchema)
+    .then(teardown)
+    .catch((err) => {
+      red(` └── ${err.message}`);
+      return Promise.bind(options).then(teardown());
+    });
+}
+
+function green(arg, ...args) {
+  console.log(chalk.green(arg), ...args);
+}
+
+function red(arg, ...args) {
+  console.error(chalk.red.bold(arg), ...args);
+}
+
+function white(arg, ...args) {
+  console.log(chalk.white(arg), ...args);
+}
+
+function processConfig(cmdArgs) {
+  // do boilerplate to get config args :/
+  const parser = new argparse.ArgumentParser({ prog: 'hz migrate' });
+
+  parser.addArgument([ 'project_path' ], {
+    default: '.',
+    nargs: '?',
+    help: 'Change to this directory before migrating',
+  });
+
+  parser.addArgument([ '--project-name', '-n' ], {
+    help: 'Name of the Horizon project server',
+  });
+
+  parser.addArgument([ '--connect', '-c' ], {
+    metavar: 'host:port',
+    default: undefined,
+    help: 'Host and port of the RethinkDB server to connect to.',
+  });
+
+  parser.addArgument([ '--rdb-user' ], {
+    default: 'admin',
+    metavar: 'USER',
+    help: 'RethinkDB User',
+  });
+
+  parser.addArgument([ '--rdb-password' ], {
+    default: undefined,
+    metavar: 'PASSWORD',
+    help: 'RethinkDB Password',
+  });
+
+  parser.addArgument([ '--start-rethinkdb' ], {
+    action: 'storeTrue',
+    help: 'Start up a RethinkDB server in the current directory'
+  });
+
+  parser.addArgument([ '--config' ], {
+    default: '.hz/config.toml',
+    help: 'Path to the config file to use, defaults to ".hz/config.toml".',
+  });
+
+  parser.addArgument([ '--skip-backup' ], {
+    action: 'storeTrue',
+    help: 'Whether to perform a backup of rethinkdb_data before migrating',
+  });
+
+  const parsed = parser.parseArgs(cmdArgs);
```

Comment:
I felt like env vars would be less useful for this. I think this implements precedence the same way as the others, config file first, then command line afterward.

I only grab a few specific items from the config file, because in general the config is for `hz serve`, we don't have any way of calling out "this option should be used when migrating only". Same thing goes for env vars.